fvm/crypto/crypto.go

```diff
@@ -37,12 +37,26 @@ func (DefaultSignatureVerifier) Verify(
 	publicKey crypto.PublicKey,
 	hashAlgo hash.HashingAlgorithm,
 ) (bool, error) {
-	hasher := NewHasher(hashAlgo)
-	if hasher == nil {
+	var hasher hash.Hasher
+
+	switch hashAlgo {
+	case hash.SHA2_256:
+		hasher = hash.NewSHA2_256()
+	case hash.SHA3_256:
+		hasher = hash.NewSHA3_256()
+	case hash.SHA2_384:
+		hasher = hash.NewSHA2_384()
+	case hash.SHA3_384:
+		hasher = hash.NewSHA3_384()
+	case hash.KMAC128:
+		hasher = crypto.NewBLSKMAC(string(tag))
+	default:
 		return false, errors.NewValueErrorf(hashAlgo.String(), "hashing algorithm type not found")
 	}
 
+	if hashAlgo != hash.KMAC128 {
 	message = append(tag, message...)
+	}
```

Comment:
I also noticed the tag here is not padded, as my GitHub ticket didn't precise it. I'll also include the fix in my upcoming branch.